panel/webserver_configuration.md

```diff
@@ -50,3 +50,54 @@ sudo ln -s /etc/apache2/sites-available/pterodactyl.conf /etc/apache2/sites-enab
 sudo a2enmod rewrite
 systemctl restart apache2
 ```
+
+## Caddy
+Caddy is a webserver with automatic https, this may come handy when you don't want to remake your ssl certificates every 90 days.
+/etc/caddy/Caddyfile:
+```
+#replace panel.yourdomain.com with your own panel domain.
+panel.yourdomain.com {
+    # Set the directory root
+    root /var/www/pterodactyl/public
+    # Let caddy know where fastcgi is for php, edit "7.2" to your php-fpm version
+    fastcgi / /run/php/php7.2-fpm.sock php
+    # rewrite all requests to index.php, this is required for how pterodactyl works
+    rewrite {
+        to {path} {path}/ /index.php
+            }
+}
+```
+
+### Running the webserver
+Once you've created the file above, simply run the command below to start the webserver manually. 
+```
+caddy -conf /etc/caddy/Caddyfile
+```
+
+To run caddy in the background use SystemD
+
+/etc/systemd/system/caddy@.service
+```
+[Unit]
+Description=Caddy HTTP/2 web server %I
+Documentation=https://caddyserver.com/docs
+After=network.target
+
+[Service]
+User=%i
+Environment=STNORESTART=yes
+ExecStart=/usr/local/bin/caddy -agree=true -conf=/etc/caddy/Caddyfile
+Restart=on-failure
+
+[Install]
+WantedBy=multi-user.target
+```
+
+```
+# reload the systemd daemon:
+systemctl daemon-reload
+# start the service:
+systemctl start caddy@root
+# enable it (to start it automatically on boot):
+systemctl enable caddy@root
```

Comment:
Why would you be starting caddy as root 🤔

Reply:
This looked the best to me as it seems to need a user and there is no good default username on linux as far as i am aware.
(didn't want to have an automatic $current-user thing)

Reply:
IIRC Apache and Nginx both create their own users to run the web server as.

Maybe just create the user for them `useradd -M -s /bin/false caddy`